src/ripple/server/impl/BaseWSPeer.h

```diff
@@ -258,6 +264,33 @@ close()
                     std::placeholders::_1)));
 }
 
+template<class Handler, class Impl>
+void
+BaseWSPeer<Handler, Impl>::
+close(const std::uint16_t& code, const std::string& reason)
+{
+    if(! strand_.running_in_this_thread())
+        return post(strand_, std::bind(
+            // disambiguate to this close overload (also above)
+            static_cast<void(BaseWSPeer::*)(const std::uint16_t&, const std::string&)>(&BaseWSPeer::close),
+            impl().shared_from_this(),
+            code, reason));
```

Comment:
Instead of `std::bind` and a `static_cast` to disambiguate the overload, what do you think about using a lambda instead? Something like:
```
    if (!strand_.running_in_this_thread())
        return post(strand_, [self = impl().shared_from_this(), reason] {
            self->close(reason);
        });
```